### DISCUSSION TITLE
How to reload an image after a failed load in Jetpack Compose?

### DISCUSSION THREAD

**Discussion by FishHawk** [coil-kt/coil#816] [Q&A] [answered]
```kotlin
val painter = rememberImagePainter(url)
Image(
    modifier = Modifier.fillMaxSize(),
    painter = painter,
    contentDescription = null,
    contentScale = ContentScale.Fit
)
when (val state = painter.state) {
    is ImagePainter.State.Error -> {
        TextButton(onClick = { }) { Text("retry") }
    }
}
```
After a failed load, the user should be able to reload the image via the retry button.
How can I do this?

**Comment by colinrtwhite** [comment, marked as answer]
It does seem like we need an API for this. As a work-around, you can do something like `painter.request = painter.request.newBuilder().setParameter("", Unit, null).build()` to restart the request.

**Reply by FishHawk** [reply]
Ok, hope the next update will solve this problem.
By the way, the setter of request is internal.
```kotlin
var request: ImageRequest by mutableStateOf(request)
    internal set
```
